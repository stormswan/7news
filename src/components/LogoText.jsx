import styled from 'styled-components'

const LogoText = styled.div.attrs ({
  size: props => props.size || '3em',
  color: props => props.color || 'white',
})`
    color: ${props => props.color};
    font-size: ${props => props.size};

    @media (max-width: 576px) {
        font-size: 1.8em;
    };

    @media (min-width: 992px) {
        font-size: 4em;
    };

    @media (min-width: 1200px) {
        font-size: 5em;
    };
`

export default LogoText