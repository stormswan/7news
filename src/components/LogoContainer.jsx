import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

import LogoText from './LogoText'
import logo from '../images/logo.png'

export default class LogoContainer extends PureComponent {
  static propTypes = {
    title: PropTypes.string.isRequired,
  }

  render() {
    return (
      <LogoContainerStyle>
        <LogoImgStyle src={logo} alt='7 news logo'/>
        <LogoText>{this.props.title}</LogoText>
      </LogoContainerStyle>
    )
  }
}



const LogoContainerStyle = styled.div`
  display: inline-flex;
  align-items:center;
  position:absolute; 
  background: rgba(25, 25, 25, .7); 
  width:100%;
  height: 32% ;
  bottom: 0; 
  left: 0;
`

const LogoImgStyle = styled.img`
  height: 100%;
`