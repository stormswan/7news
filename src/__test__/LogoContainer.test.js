import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import LogoContainer from '../components/LogoContainer'
import LogoText from '../components/LogoText'

import logoImg from '../images/logo.png'

const alt= '7 news logo'
const title = 'Home and Away'

test('Shapshot', () => {
  const logoContainer = renderer.create( <LogoContainer title='Home and Away'/>).toJSON()
  expect(logoContainer).toMatchSnapshot()
})


test('should has expected styles', () => {
  const logoContainer = renderer.create( <LogoContainer title='Home and Away' />).toJSON()
  expect(logoContainer).toHaveStyleRule('display','inline-flex')
  expect(logoContainer).toHaveStyleRule('align-items','center')
  expect(logoContainer).toHaveStyleRule('position','absolute')
  expect(logoContainer).toHaveStyleRule('background',"rgba(25, 25, 25, .7)".replace(/ /g, ''))
  expect(logoContainer).toHaveStyleRule('width','100%')
  expect(logoContainer).toHaveStyleRule('height','32%')
  expect(logoContainer).toHaveStyleRule('bottom','0')
  expect(logoContainer).toHaveStyleRule('left','0')
});


test('should work with props value', () => {
  const logoContainer = renderer.create( <LogoContainer title={title} />).root
  expect(logoContainer.findByType('img').props.alt).toBe(alt);
  expect(logoContainer.findByType('img').props.src).toBe(logoImg);
  expect(logoContainer.findByType(LogoText).props.children).toBe(title);
});