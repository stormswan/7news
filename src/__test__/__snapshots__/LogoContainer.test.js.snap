// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shapshot 1`] = `
.c2 {
  margin-left: 0.85em;
  color: white;
  font-size: 3em;
}

.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  background: rgba(25,25,25,.7);
  width: 100%;
  height: 32%;
  bottom: 0;
  left: 0;
}

.c1 {
  height: 70%;
}

@media (max-width:576px) {
  .c2 {
    font-size: 1.8em;
  }
}

@media (min-width:992px) {
  .c2 {
    font-size: 4em;
  }
}

@media (min-width:1200px) {
  .c2 {
    font-size: 5em;
  }
}

<div
  className="c0"
>
  <img
    alt="7 news logo"
    className="c1"
    src="logo.png"
  />
  <div
    className="c2"
    color="white"
    size="3em"
  >
    Home and Away
  </div>
</div>
`;
